File: Data Structures & Algorithms/validate-parentheses/submission-2.py
class Solution:
    def isValid(self, s: str) -> bool:
        par = []
        for c in s:
            if c == '(' or c == '{' or c == '[':
                par.append(c)
            elif c == ')':
                if len(par) == 0 or par.pop() != '(':
                    return False
            elif c == '}':
                if len(par) == 0 or par.pop() != '{':
                    return False
            elif c == ']':
                if len(par) == 0 or par.pop() != '[':
                    return False

        return len(par) == 0

        

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-2.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        priority_queue<int, vector<int>, std::greater<int>> q{};
        for (const auto &n : nums) {
            q.push(n);
            while (q.size() > k) q.pop();
        }
        return q.top();
    }
};
